homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -38,26 +41,28 @@ async def validate_input(hass: core.HomeAssistant, data):
         async with AjaxSession.async_create(
             data[CONF_HOST], data[CONF_USERNAME], data[CONF_PASSWORD]
         ) as ruckus:
-            system_info = await ruckus.api.get_system_info(
-                SystemStat.SYSINFO,
-            )
-            mesh_name = (await ruckus.api.get_mesh_info())[API_MESH_NAME]
+            system_info = await ruckus.api.get_system_info(SystemStat.SYSINFO)
             zd_serial = system_info[API_SYS_SYSINFO][API_SYS_SYSINFO_SERIAL]
```

Comment:
Please move this line and line 47 down out of the try... except block.

Reply:
I haven't thoroughly investigated possible combinations of responses while Unleashed is coming up (e.g. after a restart or internet outage). So I was catching KeyError in case login is functional but API calls aren't yet.

Reply:
Ok. For the future, I think the library should raise a library specific exception if an api call fails to retrieve the requested information.

Reply:
Agreed. I'll roll out a library update tomorrow which does some minimal checks, and include this when I fix the unknown error issue below.

Reply:
Can we move the lines now that only access a dict since we removed the KeyError handling?

Reply:
Sorry. Missed this one. Moved now.